detect_secrets/plugins/base.py

```diff
@@ -50,6 +50,29 @@ def secret_generator(self, string):  # pragma: no cover
         """
         pass
 
+    def adhoc_scan(self, string):
+        """To support faster discovery, we want the ability to conveniently
+        check what different plugins say regarding a single line/secret. This
+        supports that.
+
+        This is very similar to self.analyze_string, but allows the flexibility
+        for subclasses to add any other notable info (rather than just a
+        PotentialSecret type). e.g. HighEntropyStrings adds their Shannon
+        entropy in which they made their decision.
+
+        :type string: str
+        :param string: the string to analyze
+        :rtype: str
+        :returns: descriptive string that fits the format
+            <classname>: <returned-value>
+        """
+        # TODO: Handle multiple secrets on single line.
```

Comment:
Nit in general: Maybe github issue instead of `todo`'s in the code.

Reply:
(To bring attention of short-comings to reviewers, I normally call it out in github PR comment.)

Reply:
https://github.com/Yelp/detect-secrets/issues/53